src/renderers/dom/shared/wrappers/__tests__/ReactDOMInput-test.js

```diff
@@ -371,6 +371,78 @@ describe('ReactDOMInput', () => {
     expect(cNode.checked).toBe(true);
   });
 
+  it('should control radio buttons if the tree updates during render', () => {
+    var sharedParent = document.createElement('div');
+    var container1 = document.createElement('div');
+    var container2 = document.createElement('div');
+
+    sharedParent.appendChild(container1);
+
+    var aNode;
+    var bNode;
+    class ComponentA extends React.Component {
+      componentDidMount() {
+        ReactDOM.render(<ComponentB />, container2);
+      }
+      render() {
+        return (
+          <div>
+            <input
+              ref={n => aNode = n}
+              type="radio"
+              name="fruit"
+              checked={true}
+              onChange={emptyFunction}
+            />A
+          </div>
+        );
+      }
+    }
+
+    class ComponentB extends React.Component {
+      state = { changed: false };
+      handleChange = () => {
+        this.setState({
```

Comment:
Yes. Before it scanned the tree the change event callback, but `container2` isn't in the shared parent yet because this setState is batched. In the new version it doesn't scan the tree until after `componentDidUpdate` so it has time to add itself to the tree.